ledger/participant-integration-api/src/main/scala/platform/indexer/ha/PollingChecker.scala

```diff
@@ -52,19 +52,21 @@ class PollingChecker(
   // complete pool.
   def check(): Unit = synchronized {
     logger.debug(s"Checking...")
-    Try(checkBody) match {
-      case _ if closed =>
-        throw new Exception(
-          "Internal Error: This check should not be called from outside by the time the PollingChecker is closed."
-        )
+    if (closed) {
```

Comment:
Is this part of the fix or just some code shuffling?

Reply:
not directly: this is an optimization, which is possible since close and check are synchronized (if PollingChecker is closed we anyway not need to check). Rewamped it, check it out.